### PR TITLE
Catch calls to startBuildInDebugger(true) that hang builds on CI

### PR DESCRIPTION
**Pull request by big-guy**

### CODE REVIEW

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java [hunk 1]
```diff
@@ -571,6 +572,9 @@ private void calculateLauncherJvmArgs(GradleInvocation gradleInvocation) {
         buildJvmOpts.add("-ea");
 
         if (isDebug()) {
+            if (System.getenv().containsKey("CI")) {
```
Comment (ghale) [inline subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java:575]: I wonder if we should go ahead and create a `RuntimeEnvironmentDetector` or some such that encapsulates the "am I on CI?" mechanics.  That might also help with testing.

**Review by ghale — APPROVED** [review]
Pragmatically, I think this is fine as a stop gap.  I think we should also consider adding the timeout we discussed as well.

File: subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java (hunk 1 above)
Comment (big-guy) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java:575]: Maybe something as a `TestPrecondition`?

Comment (ghale) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java:575]: That would work, or just have a way to force it to consider the current environment as CI (e.g. `executor.runtimeEnvironmentDetector.forceCIEnvironment()`) so you could at least validate that "when the detector thinks I'm on CI, then I get this error".  

Stepping back, does `executer.withEnvironmentVars()` not work for testing this?

Comment (ghale) [reply subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/executer/AbstractGradleExecuter.java:575]: Oh, nvm, this is when you are configuring the executer, not when you're running it.  🙄

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by blindpirate** [comment]
@big-guy do you know what happened in the previous failing build? https://builds.gradle.org/viewLog.html?buildId=50807494

**Comment by blindpirate** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Comment by big-guy** [comment]
@bot-gradle test and merge

**Pull request merged**